bundles/org.openhab.binding.guntamatic/src/main/java/org/openhab/binding/guntamatic/internal/GuntamaticHandler.java

```diff
@@ -0,0 +1,426 @@
+                    if (type == null) {
+                        type = "";
+                    }
+
+                    if ("boolean".equals(type)) {
+                        itemType = "Switch";
+                        pattern = "";
+                    } else if ("integer".equals(type)) {
+                        pattern = "%d";
+                        if (unit.isEmpty()) {
+                            itemType = "Number";
+                        } else {
+                            itemType = guessQuantityType("Number", unit);
+                            pattern += " %unit%";
+                        }
+                    } else if ("float".equals(type)) {
+                        pattern = "%.2f";
+                        if (unit.isEmpty()) {
+                            itemType = "Number";
+                        } else {
+                            itemType = guessQuantityType("Number", unit);
+                            pattern += " %unit%";
+                        }
+                    } else if ("string".equals(type)) {
+                        itemType = "String";
+                        pattern = "";
+                    } else {
+                        if (unit.isEmpty()) {
+                            itemType = "String";
+                            pattern = "";
+                        } else {
+                            itemType = guessQuantityType("", unit);
+                            pattern = "%.2f %unit%";
+                        }
+                    }
+
+                    ChannelTypeUID channelTypeUID = new ChannelTypeUID(BINDING_ID, channel);
+                    guntamaticChannelTypeProvider.addChannelType(channelTypeUID, channel, itemType,
+                            "Guntamatic " + label, false, pattern);
+                    Channel newChannel = ChannelBuilder.create(new ChannelUID(thing.getUID(), channel), itemType)
+                            .withType(channelTypeUID).withKind(ChannelKind.STATE).withLabel(label).build();
+                    channelList.add(newChannel);
+                    channels.put(i, channel);
+                    units.put(i, unit);
+
+                    logger.debug(
+                            "Supported Channel: Idx: '{}', Name: '{}'/'{}', Type: '{}'/'{}', Unit: '{}', Pattern '{}' ",
+                            String.format("%03d", i), label, channel, type, itemType, unit, pattern);
+
+                    /*
+                     * logger.debug("Supported Channel: Idx: {}, Name: {}, Type: {}", String.format("%03d", i), channel,
+                     * itemType);
+                     */
+                }
+            }
+        }
+        ThingBuilder thingBuilder = editThing();
+        thingBuilder.withoutChannels(channelList);
+        thingBuilder.withChannels(channelList);
+        updateThing(thingBuilder.build());
+        initalized = true;
+    }
+
+    private static String guessQuantityType(String type, String unit) {
```

Comment:
Your first parameter looks simply useless.
"Number" will always be considered as type as the method is called either with "number" or "".

Reply:
thanks for the hint. I recognized it already during implemenation but needed your trigger in order to refactor it ;-)

```
    private static String guessQuantityType(String unit) {
        String quantityType = "Number";

        if ("%".equals(unit)) {
            quantityType += ":Dimensionless";
        } else if ("°C".equals(unit) || "°F".equals(unit)) {
            quantityType += ":Temperature";
        } else if ("m³".equals(unit)) {
            quantityType += ":Volume";
        } else if ("d".equals(unit) || "h".equals(unit)) {
            quantityType = += ":Time";
        } else {
            // "Number"
        }

        return quantityType;
    }
```